### ISSUE TITLE
Provide a way to replace element with new drawing instead of appending.

### ISSUE THREAD

**Issue by lgarron** [cubing/mark2#102]

**Comment by jgouly** [comment]
The easiest way, perhaps not the best, is to just change the signature of drawScramble.
`drawScramble(parentElement, state, w, h)`
to
`drawScramble(insertElement, state, w, h, append)`
Then simply `append = append == undefined ? false : append;`

**Comment by lgarron** [comment]
I know. I don't quite like that solution.
- I don't just want to tack on extra arguments for every possible option
  - If I add more options, it becomes necessary to specify explicit values for every option before.
- This requires a hardcoded, non-DRY implementation change in every scrambler file right now.
- Boolean arguments in Javascript API methods are terrible for readability.

I think the best would be to have an `options` argument that takes an object, but I'd like to have a nice way to implement it.

Also, what's the best way to clear an element? `el.innerHTML` doesn't feel very robust, even if it works.

**Comment by jgouly** [comment]
Boolean arguments in ANY API are terrible for readability :)

The interface could be changed to `drawScramble(insertElement, state, w, h, options)`
and then called with `drawScramble(insertElement, state, w, h, {append: true})`

This means having to explicitly checking if the options object as passed in, and which values were actually set, but it gives the explicit naming of optional values which is nice.

Not sure, it's what I've always used.

**Comment by lgarron** [comment]
Yeah, I think we agree on the format. I you want to implement this, just add an options handler to all the scramblers and send a pull request.

Else, I'll try to find a DRY way to do this sometime.
